internal/provider/newrelic/provider_integration_test.go

```diff
@@ -123,12 +135,205 @@ func Test_Getting_external_metric_generates_a_query_not_rejected_by_backend(t *t
 	})
 }
 
-func newrelicProviderWithMetric(t *testing.T, metric newrelic.Metric) provider.ExternalMetricsProvider {
-	t.Helper()
+// As The server must be run on port 3000 to be run locally as per the new relic client requirements, subtests
+// cannot be run in parallel.
+//
+// For more details check https://github.com/newrelic/newrelic-client-go/blob/main/pkg/region/region_constants.go
+//
+//nolint:funlen,tparallel
+func Test_does_not_hide_backend_errors(t *testing.T) {
+	t.Parallel()
 
-	testEnv := testutil.TestEnv{}
+	ctx := testutil.ContextWithDeadline(t)
+
+	testEnv := testutil.TestEnv{
+		StartKubernetes: true,
+		Region:          "Local",
+	}
 	testEnv.Generate(t)
 
+	p := newrelicProviderWithMetric(t, newrelic.Metric{
+		Query: testIntegrationQuery,
+	}, &testEnv)
+
+	runAdapter(t, &testEnv, p)
+
+	//nolint:paralleltest
+	t.Run("when_response_error", func(t *testing.T) {
+		cases := map[string]func(w http.ResponseWriter, r *http.Request){
```

Comment:
This table is looking great, thanks for covering this!

Reply:
Thanks!